examples/file_browser.md

```diff
@@ -0,0 +1,326 @@
+    while (y + 2 < DATA.length && DATA[y + 1].path.length > path.length) {
+        DATA.splice(y + 1, 1);
+    }
+}
+
+function openDir(y) {
+    const new_contents = generateDirContents(DATA[y].path.length);
+    DATA.splice(y + 1, 0, ...Array.from(new_contents));
+}
+
+function toggleDir(y) {
+    const {is_open} = DATA[y];
+    if (is_open) {
+        closeDir(y);
+    } else {
+        openDir(y);
+    }
+
+    DATA[y].is_open = !is_open;
+}
+```
+
+## Virtual Data Model
+
+`DATA` needs to be _transposed_ before we can return slices of it from our
+`dataListener()` function, because it is row-oriented and `regular-table`
+expects column-oriented data.
+
+```javascript
+function transpose(m) {
+    return m.length === 0 ? [] : m[0].map((x, i) => m.map((x) => x[i]));
+}
+```
+
+Otherwise, this `dataListener()` is very similar to `2d_array.md`.
+
+```javascript
+function dataListener(x0, y0, x1, y1) {
+    return {
+        num_rows: DATA.length,
+        num_columns: DATA[0].row.length,
+        row_headers: DATA.slice(y0, y1).map((z) => z.path.slice()),
+        column_headers: COLUMNS.slice(y0, y1),
+        data: transpose(DATA.slice(y0, y1).map(({row}) => row.slice(x0, x1))),
+    };
+}
+```
+
+## Custom Style
+
+Directory and file icon styles applied as classes, using `getMeta()`, every `td`
+is mapped back to it's row in `DATA`.
+
+```javascript
+function styleListener() {
+    for (const td of window.regularTable.querySelectorAll("tbody th")) {
+        const {y, value} = window.regularTable.getMeta(td);
+        const {row, is_open} = DATA[y];
+        const [, type] = row;
+        td.classList.toggle("fb-directory", !!value && type === "directory");
+        td.classList.toggle("fb-file", !!value && type === "file");
+        td.classList.toggle("fb-open", !!value && is_open);
+    }
+}
+```
+
+## UI
+
+When directory rows are clicked, generate new directory contents at the `td`
+metadata's `y` coordinate in `DATA` and redraw.  
+
+```javascript
+// TODO `resetAutoSize()` is not documented - this is currently required to
+// prevent the column size scroll memoize functionality from pinning the sizes
+// of the 'blank' cells, as these columns may be re-purposed as the user expands
+// or collapses the tree.  But auto-sizing is not well formalized feature yet
+// and this API is just a stand-in.
+
+function mousedownListener() {
+    if (event.target.tagName === "TH") {
+        const meta = regularTable.getMeta(event.target);
+        if (DATA[meta.y].row[1] === "directory") {
+            toggleDir(meta.y);
+            regularTable._resetAutoSize();
+            regularTable.draw();
+        }
+    }
+}
+```
+
+## Main
+
+```javascript
+function init() {
+    regularTable.setDataListener(dataListener);
+    regularTable.addStyleListener(styleListener);
+    regularTable.addEventListener("mousedown", mousedownListener);
+    regularTable.addEventListener("scroll", () => {
+        regularTable.resetAutoSize();
```

Comment:
typo, missing underscore. Should be

```
regularTable._resetAutoSize();
```

right? Super helpful that I spotted this right after you merged